sdks/python/apache_beam/coders/coders_test_common.py

```diff
@@ -570,6 +571,37 @@ def test_map_coder(self):
         }, {}, {i: str(i)
                 for i in range(5000)})
 
+  def test_sharded_key_coder(self):
+    coder = coders.ShardedKeyCoder(coders.BytesCoder())
+    # Verify cloud object representation
+    self.assertEqual({
+        '@type': 'kind:sharded_key',
+        'component_encodings': [coders.BytesCoder().as_cloud_object()]
+    },
+                     coder.as_cloud_object())
+
+    # Test binary representation
+    self.assertEqual(b'\x00\x00', coder.encode(ShardedKey(b'')))
+    self.assertEqual(b'\x00\x03key', coder.encode(ShardedKey(b'key')))
+    self.assertEqual(b'\x00\x00', coder.encode(ShardedKey(b'', b'')))
+    self.assertEqual(b'\x03123\00', coder.encode(ShardedKey(b'', b'123')))
+    self.assertEqual(b'\x03123\03key', coder.encode(ShardedKey(b'key', b'123')))
+
+    test_values = [
+        ShardedKey(b''),
```

Comment:
Have at least one example with a non-bytes key type?

Reply:
Added a string type and a tuple type.